docs/docs/kds-configuration.md

```diff
@@ -54,147 +46,102 @@ enclaveHost.start(null, null, null, kdsConfiguration) {}
+There are several supported key types, each with different trust models:
+
+- `development` - A stable master key which is not suitable for production workloads, but can be used to test KDS integration. Release mode enclaves cannot use this key type.
+- `akv_hsm` - A production ready master key which is backed by an Azure Key Vault. The key is end-to-end encrypted, however users of this key are assuming trust in Microsoft and R3.
+- `cluster` - A production ready master key which resides entirely within SGX enclaves. Neither R3 nor the KDS hosting provider have access to this key.
+
+### Defining the `policyConstraint` section
+!!!warning
+Selection of the KDS policy constraint can have significant security and operational implications.
+For this reason, users are strongly encouraged to read the following sections in their entirety.
+
+The `policyConstraint` section defines which enclaves can access a derived key.
+It is important to choose a constraint which is strict enough to prevent unauthorised access the key, but also lenient enough to allow for upgrades of enclave code and data migration if required.
+Example configurations are included later in the section.
+
+### Setting the `constraint` property
+The `constraint` property of `policyConstraint` functions in a similar manner to the [`EnclaveConstraint`](api/-conclave/com.r3.conclave.common/-enclave-constraint/index.html) object which is used when attesting to a Conclave enclave.
+When a request for a key is received by the KDS, the KDS attests that the key policy constraint matches the details of the requesting enclave before releasing a derived key.
+Just like Conclave attestation, it follows the [enclave constraints DSL](constraints.md).
+
+### Setting the `useOwnCodeHash` and `useOwnCodeSignerAndProductID` properties
+In addition to the `constraint` property, the `policyConstraint` section has two properties which allow you to include the code hash or code signer and product ID of the current enclave to the constraint.
+These parameters cannot be added to the `constraint` field directly as their values are not known at build time.
+
+- The `useOwnCodeHash` property, when set to `true`, will restrict access to the key so that it is only released to enclaves with a code hash that matches the code hash of the current enclave.
+- The `useOwnCodeSignerAndProductID` property, when set to `true`, will restrict access to the key so that it is only released to enclaves with the same code signer and product ID as the current enclave.
+
+### Example `PolicyConstraint` configurations
+#### Exact version of an enclave running on a secure platform
+This is an example of a restrictive policy constraint, whereby access to the derived key is restricted to release mode enclaves with a specific code hash running on a fully patched and up-to-date SGX capable host.
```

Comment:
In this restrictive policy constraint example, access to the derived key is restricted to release mode enclaves with a specific code hash running on a fully patched and up-to-date SGX-capable host.